java/query/query-lambda/src/test/java/sleeper/query/lambda/WarmQueryExecutorLambdaIT.java

```diff
@@ -0,0 +1,227 @@
+    }
+
+    @AfterEach
+    void tearDown() {
+        s3Client.shutdown();
+        dynamoClient.shutdown();
+        sqsClient.shutdown();
+    }
+
+    @Test
+    public void shouldCreateAQueryWithKeyTypeOfString() throws Exception {
+        // Given
+        Schema schema = getStringKeySchema();
+        TableProperties tableProperties = createTestTableProperties(instanceProperties, schema);
+        createTable(tableProperties);
+        querySerDe = new QuerySerDe(schema);
+
+        // When
+        lambda.handleRequest(new ScheduledEvent(), null);
+
+        // Then
+        ReceiveMessageResult result = sqsClient.receiveMessage(new ReceiveMessageRequest(instanceProperties.get(QUERY_QUEUE_URL)));
+        assertThat(result.getMessages()).hasSize(1);
+
+        Query query = querySerDe.fromJson(result.getMessages().get(0).getBody());
+        Query expected = buildExpectedQuery(query.getQueryId(), tableProperties.get(TABLE_NAME), schema, "a");
+
+        assertThat(query).isEqualTo(expected);
+    }
+
+    @Test
+    public void shouldCreateAQueryWitMultipleKeys() throws Exception {
+        // Given
+        Schema schema = getMultipleKeySchema();
+        TableProperties tableProperties = createTestTableProperties(instanceProperties, schema);
+        createTable(tableProperties);
+        querySerDe = new QuerySerDe(schema);
+
+        // When
+        lambda.handleRequest(new ScheduledEvent(), null);
+
+        // Then
+        ReceiveMessageResult result = sqsClient.receiveMessage(new ReceiveMessageRequest(instanceProperties.get(QUERY_QUEUE_URL)));
+        assertThat(result.getMessages()).hasSize(1);
+
+        Query query = querySerDe.fromJson(result.getMessages().get(0).getBody());
+        Query expected = buildExpectedQuery(query.getQueryId(), tableProperties.get(TABLE_NAME), schema, "a");
+
+        assertThat(query).isEqualTo(expected);
+    }
+
+    private Schema getStringKeySchema() {
+        return Schema.builder()
+                .rowKeyFields(new Field("test-key", new StringType()))
+                .sortKeyFields(new Field("test-sort", new StringType()))
+                .valueFields(new Field("test-value", new StringType()))
+                .build();
+    }
+
+    private Schema getMultipleKeySchema() {
+        return Schema.builder()
+                .rowKeyFields(List.of(new Field("test-key", new StringType()),
+                        new Field("test-key2", new StringType())))
+                .sortKeyFields(new Field("test-sort", new StringType()))
+                .valueFields(new Field("test-value", new StringType()))
+                .build();
+    }
+
+    private Query buildExpectedQuery(String id, String tableName, Schema schema, Object value) {
+        List<Range> ranges = new ArrayList<>();
+        schema.getRowKeyFields().forEach(field -> {
+            ranges.add(new Range.RangeFactory(schema)
+                    .createExactRange(field, value));
+        });
```

Comment:
This is duplicating the behaviour of the production code similarly to the other test class. We can specify which fields should have a range directly.